lib/units/dim.cpp

```diff
@@ -5,6 +5,7 @@
 #include <limits>
 #include <mutex>
 #include <shared_mutex>
+#include <stdexcept>
```

Comment:
Still needed, or leftover from debugging?

Reply:
See commit message of [8788e41](https://github.com/scipp/scipp/pull/3348/commits/8788e410662d8ed9e73f8e91625ffa8a6fd328c3)
This is needed for clang. We use `std::runtime_error` in this file but never included it. It happened to work with other compilers but that is not guaranteed.